From e6e9bb5d871720595c26776f144d539580abda80 Mon Sep 17 00:00:00 2001
From: Federico Bruni <fede@inventati.org>
Date: Mon, 31 Mar 2025 09:58:15 +0200
Subject: [PATCH] Add package data files to be included or excluded

We cannot rely on setuptools-scm to include the package
data files, because Linux distributions build from
tarballs (no git).
The list of files was previously handled by MANIFEST.in
but we can use pyproject.toml.

Use a SPDX license identifier.

Fixes #1898
---
 pyproject.toml | 15 +++++++++++++--
 1 file changed, 13 insertions(+), 2 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index 2f8c6f7c..0cc68139 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -7,7 +7,7 @@ name = "frescobaldi"
 description = "LilyPond Music Editor"
 authors = [{name = "Wilbert Berendsen", email = "info@frescobaldi.org"}]
 maintainers = [{name = "Wilbert Berendsen", email = "info@frescobaldi.org"}]
-license = {text = "GPL"}
+license = {text = "GPL-2.0-or-later"}
 requires-python = ">=3.8"
 dependencies = ["pyqt6>=6.6", "python-ly >= 0.9.5", "qpageview>=1.0.0"]
 dynamic = ["version"]
@@ -17,7 +17,6 @@ classifiers = [
     "Environment :: Win32 (MS Windows)",
     "Environment :: X11 Applications :: Qt",
     "Intended Audience :: End Users/Desktop",
-    "License :: OSI Approved :: GNU General Public License (GPL)",
     # Natural Language :: Chinese (Hong Kong) is not yet accepted by pypi
     #"Natural Language :: Chinese (Hong Kong)",
     "Natural Language :: Chinese (Simplified)",
@@ -71,6 +70,18 @@ include = ["frescobaldi*"]
 [tool.setuptools.dynamic.version]
 attr = "frescobaldi.appinfo.version"
 
+# Files not added by default in the wheel but needed on runtime.
+[tool.setuptools.package-data]
+"*" = [
+  "*.dic", "*.ily", "*.js", "*.ly", "*.md", "*.mo", "*.png", "*.svg",
+  "frescobaldi.1", "LICENSE", "THANKS", "TODO",
+  "linux/org.frescobaldi.Frescobaldi.desktop",
+  "linux/org.frescobaldi.Frescobaldi.metainfo.xml"
+]
+
+[tool.setuptools.exclude-package-data]
+"*" = ["*~", ".git*"]
+
 [tool.briefcase]
 project_name = "Frescobaldi"
 bundle = "org.frescobaldi"
-- 
2.49.0

